### PR TITLE
Update modification date on object changes

### PR DESCRIPTION
**Pull request by pl-buiquang**
Signed-off-by: Paul Bui-Quang <paul.buiquang@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
feature

**What is the current behavior?** *(You can also link to an open issue here)*
the modification date of the afs object isn't updated on object changes (content or dep changes)

**What is the new behavior (if this is a feature change)?**
the modification date is updated over various changes depending of the object type

### CODE REVIEW

File: security-analysis/security-analysis-afs-local/src/main/java/com/powsybl/security/afs/local/LocalSecurityAnalysisRunningService.java [hunk 1]
```diff
@@ -78,6 +78,7 @@ public void run(SecurityAnalysisRunner runner) {
                         return null;
                     }).join();
         } finally {
+            runner.updateModificationTime();
```
Comment (sylvlecl) [inline security-analysis/security-analysis-afs-local/src/main/java/com/powsybl/security/afs/local/LocalSecurityAnalysisRunningService.java]: I wonder : do we want to update modification time even when no result is written ?
Maybe it's only the "writeResult" which should update the modification time ?

From a broader point of view, maybe the writeBinaryData method should actually be responsible for updating the modification date ?
(not part of this PR of course)

Comment (pl-buiquang) [reply security-analysis/security-analysis-afs-local/src/main/java/com/powsybl/security/afs/local/LocalSecurityAnalysisRunningService.java]: hm.. I guess your approach seems better. the AppStorage should be in charge of updating modification time when it modifies data (writeBinaryData, operation on timeseries and dependencies)

Comment (pl-buiquang) [reply security-analysis/security-analysis-afs-local/src/main/java/com/powsybl/security/afs/local/LocalSecurityAnalysisRunningService.java]: then the updateModificationTime method should not even be accessible since appstorage is  the sole responsable for updating afs objects

Comment (sylvlecl) [reply security-analysis/security-analysis-afs-local/src/main/java/com/powsybl/security/afs/local/LocalSecurityAnalysisRunningService.java]: "Pas faux" :)

Comment (pl-buiquang) [reply security-analysis/security-analysis-afs-local/src/main/java/com/powsybl/security/afs/local/LocalSecurityAnalysisRunningService.java]: what do you think @geofjamg ?

**Comment by mathbagu** [comment]
This PR has to be moved to powsybl-afs
